path_login.go

```diff
@@ -92,52 +91,58 @@ func (b *backend) operationLoginUpdate(ctx context.Context, req *logical.Request
 		return logical.ErrorResponse(err.Error()), nil
 	}
 
-	// Ensure the time it was signed is no more than 5 minutes in the past
-	// or 30 seconds in the future. This is a guard against some replay attacks.
-	fiveMinutesAgo := timeReceived.Add(time.Minute * time.Duration(-5))
-	thirtySecondsFromNow := timeReceived.Add(time.Second * time.Duration(30))
-	if signingTime.Before(fiveMinutesAgo) {
-		return logical.ErrorResponse(fmt.Sprintf("request is too old; signed at %s but received request at %s; raw signing time is %s", signingTime, timeReceived, signingTimeRaw)), nil
-	}
-	if signingTime.After(thirtySecondsFromNow) {
-		return logical.ErrorResponse(fmt.Sprintf("request is too far in the future; signed at %s but received request at %s; raw signing time is %s", signingTime, timeReceived, signingTimeRaw)), nil
-	}
-
-	// Ensure the private key used to create the signature matches our client
-	// certificate, and that it signed the same data as is presented in the body.
-	// This offers some protection against MITM attacks.
-	matchingCert, err := signatures.Verify(signature, &signatures.SignatureData{
-		SigningTime: signingTime,
-		Role:        roleName,
-		Certificate: clientCertificate,
-	})
-	if err != nil {
-		return logical.ErrorResponse(err.Error()), nil
-	}
-
-	// Ensure the matching certificate was actually issued by the CA configured.
-	// This protects against self-generated client certificates.
 	config, err := config(ctx, req.Storage)
 	if err != nil {
 		return nil, err
 	}
 	if config == nil {
 		return nil, errors.New("no CA is configured for verifying client certificates")
 	}
-	verifyOpts, err := config.VerifyOpts()
+
+	// Ensure the time it was signed isn't too far in the past or future.
+	oldestAllowableSigningTime := timeReceived.Add(-1 * config.LoginMaxSecOld)
+	furthestFutureAllowableSigningTime := timeReceived.Add(config.LoginMaxSecAhead)
+	if signingTime.Before(oldestAllowableSigningTime) {
+		return logical.ErrorResponse(fmt.Sprintf("request is too old; signed at %s but received request at %s; allowable seconds old is %d", signingTime, timeReceived, config.LoginMaxSecOld)), nil
```

Comment:
Will times like `config.LoginMaxSecOld` print correctly with `%d`?

Reply:
Oh, right, it will print out the nanoseconds rather than the seconds. Looks like I need to do [this](https://play.golang.org/p/LFPz6uMJc4S). I'll push an update.